framework/src/auxkernels/SelfAux.C

```diff
@@ -19,13 +19,19 @@ SelfAux::validParams()
   InputParameters params = AuxKernel::validParams();
   params.addClassDescription(
       "Returns the specified variable as an auxiliary variable with the same value.");
+  params.addCoupledVar("v",
+                       "Optional variable to take the value of. If omitted the value of the "
+                       "`variable` itself is returned.");
   return params;
 }
 
-SelfAux::SelfAux(const InputParameters & parameters) : AuxKernel(parameters) {}
+SelfAux::SelfAux(const InputParameters & parameters)
+  : AuxKernel(parameters), _v(isCoupled("v") ? coupledValue("v") : _u)
```

Comment:
I have not used `SelfAux` before. For the original implementation, `_u` is the `auxvariable` name? This `AuxKernel` sets the `Auxvariable` to itself value? I guess I did not quite get when we need this.

Reply:
TBH I don't know what this AuxKernel was designed for. I just extended it, because we didn't seem to have an AuxKernel that did what I needed and this one was closest. To clarify I needed to create a copy of a given AuxVariable with a different name to enable exodiffing between to files that only diffed on the name of one variable.